src/main/scala/io/pinecone/spark/pinecone/PineconeDataWriter.scala

```diff
@@ -62,7 +57,13 @@ case class PineconeDataWriter(
       if (!record.isNullAt(4)) {
         val sparseVectorStruct = record.getStruct(4, 2)
         if (!sparseVectorStruct.isNullAt(0) && !sparseVectorStruct.isNullAt(1)) {
-          val sparseId = sparseVectorStruct.getArray(0).toIntArray().map(int2Integer).toIterable
+          val sparseIndices = sparseVectorStruct.getArray(0).toLongArray()
+
+          if (sparseIndices.exists(index => index < 0 || index > 0xFFFFFFFFL)) {
```

Comment:
Reject everything outside of the range of unsigned 32-bit int for sparse indices

Reply:
Totally, let me update it!!